### PR TITLE
[ICC-252] Manage opt-in for  Reminder notifications

### PR DESCRIPTION
**Pull request by gquadrati**
#### List of Changes

- [update fn-commons](https://github.com/pagopa/io-functions-service-messages/commit/c0aabaa7cf836df0f333da841b9088df07d7633c)
- [add UserProfileReader and implementation](https://github.com/pagopa/io-functions-service-messages/commit/aae2ed9ca73ad7e27288b590c8b29942749a48c6)
- [update Notify handler](https://github.com/pagopa/io-functions-service-messages/commit/c5242ad2d5747f0b93f4b522e9b958fbe9397f5e)
- [add unit tests](https://github.com/pagopa/io-functions-service-messages/pull/23/commits/fa8f5f1b48ac839265d60cbb9d43b502af84885c)
- [[#ICC-252] prepare integration tests with io-backend mock](https://github.com/pagopa/io-functions-service-messages/pull/23/commits/1198d1daaa5f54cb5ffe1cdf4d48c9684a99836b)
- [[#ICC-252] add integration tests](https://github.com/pagopa/io-functions-service-messages/pull/23/commits/a48c58c6f87ac4307a40b35048c1fb4ebd0ae21b)

#### Motivation and Context

send Reminder notification only when user has given his consent.

#### How Has This Been Tested?

Unit tests

#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: Notify/handler.ts [hunk 1]
```diff
@@ -76,14 +84,17 @@ const canSendReminderNotification = (
...
+  retrieveUserProfile: UserProfileReader
+) => (
   notificationType: NotificationType,
   fiscalCode: FiscalCode
 ): TE.TaskEither<Error, boolean> =>
   match(notificationType)
     .when(isReminderNotification, _ =>
       isBetaTester(fiscalCode)
-        ? canSendReminderNotification(fiscalCode)
+        ? canSendReminderNotification(retrieveUserProfile)(fiscalCode)
```
Comment (AleDore) [inline Notify/handler.ts:97]: What about use `Boolean.fold` here ?

**Review by AleDore — COMMENTED** [review]
Aside from a little style's consideration, can we add some integration tests?

File: Notify/handler.ts (hunk 1 above)
Comment (gquadrati) [reply Notify/handler.ts:97]: We're going to remove it, so we can avoid refactor right now

**Review by fabriziopapi — APPROVED** [review]
LGTM, but we really need a "feature flag" library.

File: __integrations__/__mocks__/fixtures.ts [hunk 2]
```diff
@@ -339,6 +346,28 @@ export const fillServices = async (
...
 };
 
+export const fillProfiles = async (
+  db: Database,
+  profiles: ReadonlyArray<ProfileCollection.Profile>
+): Promise<void> => {
+  await pipe(
+    db.container(ProfileCollection.PROFILE_COLLECTION_NAME),
+    TE.of,
+    TE.map(p => new ProfileCollection.ProfileModel(p)),
```
Comment (BurnedMarshal) [inline __integrations__/__mocks__/fixtures.ts:356]: A more compact version:
```suggestion
    p => TE.of(new ProfileCollection.ProfileModel(p)),
```
I see the same pattern on row 376

File: __integrations__/__mocks__/mock.profiles.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    fiscalCode: anAutoFiscalCodeWithReminderEnabled,
+    reminderStatus: ReminderStatusEnum.ENABLED
+  },
+  {
+    fiscalCode: anAutoFiscalCodeWithReminderDisabled,
+    reminderStatus: ReminderStatusEnum.DISABLED
+  },
+  {
+    fiscalCode: anAutoFiscalCodeWithReminderNotDefined
+  }
```
Comment (BurnedMarshal) [inline __integrations__/__mocks__/mock.profiles.ts:47]: the profile coming from the model cannot has `reminderStatus === undefined`. Undefined values on database are converted to `UNSET`.
```suggestion
  {
    fiscalCode: anAutoFiscalCodeWithReminderNotDefined,
    reminderStatus: "UNSET"
  }
```

**Review by BurnedMarshal — COMMENTED** [review]
LGMT, only a minor change on empty reminderStatus

File: __integrations__/__mocks__/mock.profiles.ts (hunk 3 above)
Comment (gquadrati) [reply __integrations__/__mocks__/mock.profiles.ts:47]: Here we are populating the database, so it's fine not having `reminderStatus` set, it's up to the decoder setting it to `UNSET` while retrieving it from the endpoint.
Indeed we could have add another test with `reminderStatus` explicitly set to `UNSET`.

**Pull request merged**